vortex-dtype/src/dtype.rs

```diff
@@ -133,14 +133,16 @@ impl Display for DType {
                 n
             ),
             List(edt, n) => write!(f, "list({}){}", edt, n),
-            Extension(ext, n) => write!(
+            Extension(ext) => write!(
                 f,
-                "ext({}{}){}",
+                "ext({}, {}{}){}",
                 ext.id(),
+                ext.storage_dtype()
+                    .with_nullability(Nullability::NonNullable),
                 ext.metadata()
                     .map(|m| format!(", {:?}", m))
                     .unwrap_or_else(|| "".to_string()),
-                n
+                ext.storage_dtype().nullability(),
```

Comment:
the nullability split thing here might read weird (it will say NonNullable above... and then maybe-Nullable here?)